### PR TITLE
Fix ds nvme

### LINKED ISSUE #34429: deepspeed zero3 NVMe offload is  not working
### System Info

transformers==4.46.0
accelerate==1.0.1
sentencepiece==0.2.0
deepspeed==0.15.3

pin

### Who can help?

@muellerz @SunMarc @ArthurZucker @amyeroberts @qubvel

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

accelerate configr:
compute_environment: LOCAL_MACHINE                                                                                                                                           
debug: false
deepspeed_config:
  deepspeed_multinode_launcher: standard
  gradient_accumulation_steps: auto
  gradient_clipping: 1.0
  offload_optimizer_device: nvme
  offload_param_device: nvme
  zero3_init_flag: true
  zero3_save_16bit_model: false
  zero_stage: 3
distributed_type: DEEPSPEED
downcast_bf16: 'no'
machine_rank: 0
main_process_ip: 0.0.0.0
main_process_port: 0
main_training_function: main
mixed_precision: bf16
num_machines: 3
num_processes: 24
rdzv_backend: c10d
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false

================================================

I launch the code using 

```
accelerate launch --config_file config.yml code.py
```

```
from transformers import AutoModelForVision2Seq, BitsAndBytesConfig

model_kwgs = {
     "pretrained_model_name_or_path": "local_folder/contains/paligemma-3b-pt-896",
     "trust_remote_code":True,
     }

model = AutoModelForVision2Seq.from_pretrained(**model_kwgs)
```

### Expected behavior

Have a NVMe offloaded model.

### PR DESCRIPTION
**Pull request by eljandoubi**
# What does this PR do?

Fixes #34429 

## Who can review?

Models:

- text models: @ArthurZucker
- vision models: @amyeroberts, @qubvel

Integrations:

- deepspeed: HF Trainer/Accelerate: @muellerzr

### CODE REVIEW

**Comment by eljandoubi** [comment]
@SunMarc @muellerzr Do you have any feedback?

**Review by SunMarc — COMMENTED** [review]
Hey @eljandoubi , thanks for the contribution. Could you explain why this is needed ? Thanks !

**Comment by eljandoubi** [comment]
Hey @SunMarc. When using NVMe offloading, recursively calling `deepspeed.zero.Init` causes a pinning error.  Because Transformers and accelerate do not support multiple models, `deepspeed.zero.Init` should be called at most once per process.

**Comment by SunMarc** [comment]
Thanks for the additional context ! I think that we do support multiple models with deepspeed https://huggingface.co/docs/accelerate/en/usage_guides/deepspeed_multiple_model . cc @muellerzr

**Comment by eljandoubi** [comment]
Anyway, we might use a context manager solution to prevent embedded calls to DeepSpeed.zero.Init.

**Comment by eljandoubi** [comment]
@ArthurZucker @SunMarc @muellerzr Is there any update on the pull request?

**Review by SunMarc — APPROVED** [review]
LGTM ! I think you pushed some unrelated changes. Could you revert those ?

**Comment by eljandoubi** [comment]
@SunMarc can you underscore them?

File: examples/research_projects/lxmert/demo.ipynb
Comment (ArthurZucker) [inline examples/research_projects/lxmert/demo.ipynb:1]: this and all ipynb are unrelated

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -223,6 +224,15 @@ def set_quantized_state():
     finally:
         _is_quantized = False
 
+@contextmanager
+def set_zero3_state():
```
Comment (ArthurZucker) [inline src/transformers/modeling_utils.py:228]: can we have a small comment documentation about that (explaning the issue we fixed with this basically) 😉

**Review by ArthurZucker — COMMENTED** [review]
Thanks, let's revert notebook changes, and add a one liner about what we are adding!

**Review by muellerzr — COMMENTED** [review]
Thanks! Same note as Arthur, otherwise LG2M!

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗

**Comment by ArthurZucker** [comment]
Just missing `make style`

**Pull request merged**